### DISCUSSION TITLE
useEffect fires even when the session is undefined

### DISCUSSION THREAD

**Discussion by tessamerrill** [nextauthjs/next-auth#2258] [Help] [answered]
Hi there, I have this code that should run when the session / session user is defined:
```js
  useEffect(() => {
    console.log('authed')
    const { user: userData } = useUser(session);
    setUser(userData);
  }, [session.user]);
```

However, in `useUser()`, it returns `undefined` as the session.

```js
function useUser(session: Session) {
  console.log(session) // undefined
  const db = firebase.firestore();

  db.doc(`accounts/${session.userId}`).get().then((userData) => {
    console.log(userData.data())
    setUser(userData.data());
  });

  return { user };
}
```

(let me know if this isn't a next-auth issue)

What do I do, and how can I fix it?

Thanks!

**Comment by balazsorban44** [comment, marked as answer]
It's a React hook issue. Have a read on https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level 😊

Update: Maybe I misunderstood, but be careful with the naming of your functions. `use` is usually reserved for React hooks. The problem is still not next-auth specific though. you will need an early return when session is undefined
